### ISSUE TITLE
Overlay: cannot keep the focus trapped in a modal

### ISSUE THREAD

**Issue by dorina1** [wet-boew/wet-boew#8854]
When we Shift-TAB in any modals, the focus does not stay in the modal, but it goes behind into the browser top elements and rotates until you press TAB and you go back into the modal: [Overlay](https://wet-boew.github.io/v4.0-ci/demos/overlay/overlay-en.html)

Similar behavior when we TAB, the focus does not stay in the modal : it goes behind and after a few more tabs it comes back in. 

This occurs in multiple browsers (Edge, IE11, Chrome).

Keyboard-access expected behavior for modals should be:
**TAB**
- Moves focus to next focusable element inside the modal.
- When focus is on the last focusable element in the modal, moves focus to the first focusable element in the modal.
**SHIFT-TAB**
- Moves focus to previous focusable element inside the modal.
When focus is on the first focusable element in the dialog, moves focus to the last focusable element in the modal.
**ESC**
- Closes the modal (this works)

**Comment by EricDunsworth** [comment]
This will finally be fixed by #10153.

Looks like it was previously addressed in the lightbox plugin in #8855 (which also benefited the [overlay demo page](https://wet-boew.github.io/v4.0-ci/demos/overlay/overlay-en.html)'s lightbox examples).

**PS:**
I was unable to replicate this behaviour:
> Similar behavior when we TAB, the focus does not stay in the modal : it goes behind and after a few more tabs it comes back in.

Maybe it got fixed at some point between when this issue was originally posted and now?

The overlay plugin previously did a poor job of keeping track of interactive ("tabbable") elements within its container due to buggy logic it borrowed from a super old version of jQuery UI. It caused problems when trying to forward-tab beyond the overlay's X close button _if_ the first interactive element in the container wasn't reachable. That'll be fixed by #10152.

**PS2:**
The OP's expected behaviours correspond to what I'm getting with both of the aforementioned PRs in place 🤓.
